kotsadm/pkg/store/ocistore/version_store.go

```diff
@@ -331,49 +331,46 @@ func (s OCIStore) CreateAppVersion(appID string, currentSequence *int64, appName
+		// will support multiple downstreams, so this is cleaner here for now
+		licenseSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "License")
+		if err != nil {
+			return int64(0), errors.Wrap(err, "failed to marshal license spec")
+		}
+		configSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Config")
+		if err != nil {
+			return int64(0), errors.Wrap(err, "failed to marshal config spec")
+		}
+		configValuesSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "ConfigValues")
+		if err != nil {
+			return int64(0), errors.Wrap(err, "failed to marshal configvalues spec")
 		}
 
-		if currentSequence != nil {
-			// there's a small chance this is not optimal, but no current code path
-			// will support multiple downstreams, so this is cleaner here for now
-			licenseSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "License")
-			if err != nil {
-				return int64(0), errors.Wrap(err, "failed to marshal license spec")
-			}
-			configSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Config")
-			if err != nil {
-				return int64(0), errors.Wrap(err, "failed to marshal config spec")
-			}
-			configValuesSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "ConfigValues")
-			if err != nil {
-				return int64(0), errors.Wrap(err, "failed to marshal configvalues spec")
-			}
+		configOpts := kotsconfig.ConfigOptions{
+			ConfigSpec:       configSpec,
+			ConfigValuesSpec: configValuesSpec,
+			LicenseSpec:      licenseSpec,
+		}
+		if registryInfo != nil {
+			configOpts.RegistryHost = registryInfo.Hostname
+			configOpts.RegistryNamespace = registryInfo.Namespace
+			configOpts.RegistryUser = registryInfo.Username
+			configOpts.RegistryPassword = registryInfo.Password
+		}
 
-			configOpts := kotsconfig.ConfigOptions{
-				ConfigSpec:       configSpec,
-				ConfigValuesSpec: configValuesSpec,
-				LicenseSpec:      licenseSpec,
-			}
-			if registryInfo != nil {
-				configOpts.RegistryHost = registryInfo.Hostname
-				configOpts.RegistryNamespace = registryInfo.Namespace
-				configOpts.RegistryUser = registryInfo.Username
-				configOpts.RegistryPassword = registryInfo.Password
-			}
+		// check if version needs additional configuration
+		needsConfig, err := kotsconfig.NeedsConfiguration(configOpts)
+		if err != nil {
+			return int64(0), errors.Wrap(err, "failed to check if version needs configuration")
+		}
 
-			// check if version needs additional configuration
-			t, err := kotsconfig.NeedsConfiguration(configOpts)
-			if err != nil {
-				return int64(0), errors.Wrap(err, "failed to check if version needs configuration")
-			}
-			if t {
-				downstreamStatus = "pending_config"
-			}
+		downstreamStatus := "pending"
+		if currentSequence == nil && needsConfig {
+			downstreamStatus = "pending_config"
+		} else if kotsKinds.Preflight != nil && !skipPreflights {
+			downstreamStatus = "pending_preflight"
+		} else if currentSequence != nil && needsConfig {
```

Comment:
i think this should come before the preflight condition. probably the first if statement should just check for `needsConfig`.

Reply:
In the original logic it looks like `pending_config` overrides `pending_preflight`.  So I kept that.